corehq/apps/reports/tasks.py

```diff
@@ -337,11 +338,33 @@ def extract_form_info(form, properties=None, case_ids=case_ids):
 
         return form_info
 
-    key = [domain, app_id, xmlns]
-    form_ids = {f['id'] for f in XFormInstance.get_db().view("attachments/attachments",
-                                                             start_key=key + [startdate],
-                                                             end_key=key + [enddate, {}],
-                                                             reduce=False)}
+    if not export_is_legacy:
+        query = (FormES()
+                 .domain(domain)
+                 .app(app_id)
+                 .xmlns(xmlns)
+                 .submitted(gte=parse(startdate), lte=parse(enddate))
+                 .remove_default_filter("has_user"))
+        form_ids = set()
+        for form in query.scroll():
+            try:
+                for attachment in form['_attachments'].values():
```

Comment:
@snopoke Looks like a [query string](https://www.elastic.co/guide/en/elasticsearch/reference/1.7/query-dsl-query-string-query.html) would allow a wildcard search on the field, but since we have `'dynamic': False` for the `_attachments` field in the xform mapping, it's not searchable. So one option would be to index that object. I suspect it would also be possible to write a script query that would work, but haven't looked too far into that yet.

I'm not sure if it's worth spending the time on though. For some form exports, every xform will have an image anyways (like if an image question is required), so adding that filter won't reduce the number of hits on the query in the worst case. Do you think it's still worth looking into ES scripts?

I'll definitely limit the fields returned, that's an easy win for sure.